"Put the license information" fails

I'm having an issue related to #1084:
```
[INFO ]  Restoring state at step 'finish', as requested.
[INFO ]  =================================================================
[INFO ]  Finalizing the toolchain's directory
[INFO ]    Stripping all toolchain executables
[EXTRA]    Installing the populate helper
[EXTRA]    Installing a cross-ldd helper
[EXTRA]    Creating toolchain aliases
[EXTRA]    Removing installed documentation
[EXTRA]    Collect license information from: /Volumes/ldev/powerpc64le-unknown-linux-gnu-build/.build/powerpc64le-unknown-linux-gnu/src
[EXTRA]    Put the license information to: /Volumes/ldev/x-tools/powerpc64le-unknown-linux-gnu/share/licenses
[ERROR]
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Finalizing the toolchain's directory'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@374]
[ERROR]  >>        called from: CT_InstallCopyingInformation[scripts/functions@2428]
[ERROR]  >>        called from: do_finish[scripts/build/internals.sh@148]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@688]


Current command:
   'cp' '-av' '/Volumes/ldev/powerpc64le-unknown-linux-gnu-build/licenses.d' '/Volumes/ldev/x-tools/powerpc64le-unknown-linux-gnu/share/licenses/crosstool-ng/'
exited with error code: 1
Please fix it up and finish by exiting the shell with one of these values:
    1  fixed, continue with next build command
    2  repeat this build command
    3  abort build

ct-ng:/Volumes/ldev/powerpc64le-unknown-linux-gnu-build> echo $CT_PREFIX_DIR

ct-ng:/Volumes/ldev/powerpc64le-unknown-linux-gnu-build> echo $CT_TOP_DIR
/Volumes/ldev/powerpc64le-unknown-linux-gnu-build

ct-ng:/Volumes/ldev/powerpc64le-unknown-linux-gnu-build> echo $CT_PREFIX
/Volumes/ldev/x-tools

ct-ng:/Volumes/ldev/powerpc64le-unknown-linux-gnu-build> exit 3
exit
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      https://crosstool-ng.github.io/docs/known-issues/
[ERROR]  >>
[ERROR]  >>  If you feel this is a bug in crosstool-NG, report it at:
[ERROR]  >>      https://github.com/crosstool-ng/crosstool-ng/issues/
[ERROR]  >>
[ERROR]  >>  Make sure your report includes all the information pertinent to this issue.
[ERROR]  >>  Read the bug reporting guidelines here:
[ERROR]  >>      http://crosstool-ng.github.io/support/
[ERROR]
[ERROR]  (elapsed: 97:17.00)
[02:30] / gmake[1]: *** [/usr/local/bin/ct-ng:235: build] Error 1
gmake: *** [/usr/local/bin/ct-ng:287: finish] Error 2
```
I'm running `ct-ng build` with `CT_PREFIX` set to a directory on case sensitive filesystem: `/Volumes/ldev/x-tools`. `.config` is `powerpc64le-unknown-linux-gnu` plus ct-ng debugging.

`CT_TOP_DIR` contains path to the directory where I have `.config` and where I start `ct-ng build`. So I'm not sure if https://github.com/crosstool-ng/crosstool-ng/pull/1084/commits/aec70d53102fcd5f468dc8439a25010254166c22#diff-f4677383f012769a82f4885f7d508cb6R2422 should work at all. Also, the files in `licenses.d` are not added to `Automake.am`.

- [x] crosstool-NG version:
1.23.0.523-61687
- [x] host OS and its version
Darwin Kernel Version 18.2.0
- [x] [.config](https://github.com/crosstool-ng/crosstool-ng/files/2566721/config.txt)
- [x] [build.log.xz](https://github.com/crosstool-ng/crosstool-ng/files/2566728/build.log.xz.txt)

--- a/Makefile.am
+++ b/Makefile.am
@@ -67,6 +67,8 @@ install-data-hook:
 	gzip -9 $(DESTDIR)$(man1dir)/$(ctng_progname).1
 	chmod a+x $(DESTDIR)$(pkgdatadir)/scripts/config.guess
 	chmod a+x $(DESTDIR)$(pkgdatadir)/scripts/config.sub
+	rm -f $(DESTDIR)$(pkgdatadir)/LICENSE
+	ln -s COPYING $(DESTDIR)$(pkgdatadir)/LICENSE
 if INSTALL_BASH_COMPLETION
 	chmod a-x $(DESTDIR)$(compdir)/$(ctng_progname)
 endif

--- a/bootstrap
+++ b/bootstrap
@@ -742,7 +742,7 @@ msg "*** Gathering the list of data files to install"
 {
     declare -A seen_files
     echo -n "verbatim_data ="
-    find config contrib packages samples scripts -type f | LANG=C sort | while read f; do
+    find COPYING config contrib licenses.d packages samples scripts -type f | LANG=C sort | while read f; do
         # Implement some kind of .installignore for these files?
         case "${f}" in
             # Skip certain files not needed at runtime (used above for generating kconfig

--- a/ct-ng.in
+++ b/ct-ng.in
@@ -20,6 +20,7 @@ ifeq (@enable_local@,yes)
 export CT_LIB_DIR:=$(patsubst %/,%,$(dir $(CT_NG)))
 export CT_LIBEXEC_DIR:=$(CT_LIB_DIR)/kconfig
 export CT_DOC_DIR:=$(CT_LIB_DIR)/docs
+export CT_LICENSE_DIR
 else
 # Paths and values set by ./configure
 # Don't bother to change it other than with a new ./configure!

--- a/scripts/functions
+++ b/scripts/functions
@@ -2402,27 +2402,21 @@ fi
 
 CT_InstallCopyingInformation()
 {
-    local pkgname
-    local pkgdir
     local licfile
     local dstdir
 
     CT_DoLog EXTRA "Collect license information from: ${CT_SRC_DIR}"
     CT_DoLog EXTRA "Put the license information to: ${CT_PREFIX_DIR}/share/licenses"
 
-    shopt -s nullglob
-
-    for pkgdir in ${CT_SRC_DIR}/*; do
-        pkgname=$(basename "${pkgdir}")
-        for licfile in ${pkgdir}/{COPYING*,LICENSE*}; do
-            dstdir="${CT_PREFIX_DIR}/share/licenses/${pkgname}"
-            mkdir -p "${dstdir}"
-            CT_DoExecLog ALL cp -av "${licfile}" "${dstdir}/"
-        done
+    for licfile in $( find "${CT_SRC_DIR}" -follow -type f -a \( -name "COPYING*" -o -name "LICENSE*" \) ); do
+        dstdir="${licfile%/*}"
+        dstdir="${CT_PREFIX_DIR}/share/licenses${dstdir#${CT_SRC_DIR}}"
+        mkdir -p "${dstdir}"
+        CT_DoExecLog ALL cp -av "${licfile}" "${dstdir}/"
     done
 
     # Also add crosstool's information
-    for licfile in ${CT_TOP_DIR}/{COPYING*,LICENSE*,licenses.d}; do
+    for licfile in ${CT_LIB_DIR}/{COPYING*,LICENSE*,licenses.d}; do
         dstdir="${CT_PREFIX_DIR}/share/licenses/crosstool-ng"
         mkdir -p "${dstdir}"
         CT_DoExecLog ALL cp -av "${licfile}" "${dstdir}/"

--- a/scripts/scripts.mk
+++ b/scripts/scripts.mk
@@ -31,11 +31,11 @@ ifneq ($(strip $(V)),2)
   curl_silent_opt = --silent
 endif
 
-ifneq ($(CT_wget),)
-download_cmd = $(CT_wget) --passive-ftp $(wget_silent_opt) -O $@
+ifneq ($(CT_WGET),)
+download_cmd = $(CT_WGET) --passive-ftp $(wget_silent_opt) -O $@
 else
-ifneq ($(CT_curl),)
-download_cmd = $(CT_curl) --ftp-pasv $(curl_silent_opt) -o $@
+ifneq ($(CT_CURL),)
+download_cmd = $(CT_CURL) --ftp-pasv $(curl_silent_opt) -o $@
 else
 download_cmd = $(error wget or curl needed for downloads)
 endif

